rest/model/user.go

```diff
@@ -204,22 +153,28 @@ func (n *APINotificationPreferences) ToService() (interface{}, error) {
 	spawnHostOutcome := utility.FromStringPtr(n.SpawnHostOutcome)
 	commitQueue := utility.FromStringPtr(n.CommitQueue)
 	if !user.IsValidSubscriptionPreference(buildBreak) {
-		return nil, errors.Errorf("invalid build break subscription preference '%s'", buildBreak)
+		return user.NotificationPreferences{},
```

Comment:
Can this be validated separately as well?

Reply:
I started removing this but I think it actually makes sense here. If these values don't match the values we have defined for UserSubscriptionPreference, then we would have an issue. I'm okay with some BuildFromService/ToService functions still returning errors, since I think the nice thing about not having these functions implement an interface is that they can be what we need them to be. I'll refactor this to make it cleaner though.

Reply:
Perhaps we'd want to name them something like `BuildFromServiceAndValidate` in these cases where we can return an error. Although I'm not extremely opinionated about it

Reply:
I think I'm fine with the error since it's more that we just _can't_ buildFromService if the types aren't right, in this case, not that the input isn't "correct" in terms of what we want.